extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -333,7 +359,14 @@ export class AmpConsent extends AMP.BaseElement {
       for (let i = 0; i < iframes.length; i++) {
         if (iframes[i].contentWindow === event.source) {
           const action = data['action'];
-          this.handleAction_(action, consentString, metadata);
+          // Check if we have a valid action and valid state
+          if (
+            isEnumValue(ACTION_TYPE, action) &&
+            this.isReadyToHandleAction_()
+          ) {
+            // TODO(micajuineho): Set purpose consent
+            this.handleAction_(action, consentString, metadata);
+          }
           return;
```

Comment:
This dangling `return` is throwing me off. I suspect a bug here.

Reply:
Yeah, not sure why it was there before. Removing now.